### PR TITLE
Network Manager - Add Devices, Links & Sites

### LINKED ISSUE #7844: Network Manager: Add additional endpoints for Sites, Links & Devices
It would be useful to add support for the endpoints to Network Manager:
- Devices
- Links
- Sites

The actions would include **Create**, **Get** & **Delete** for each endpoint.

https://docs.aws.amazon.com/networkmanager/latest/APIReference/API_GetDevices.html
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/networkmanager/client/get_sites.html

### PR DESCRIPTION
**Pull request by zkarpinski**
Resolves #7844 

Network Manager Endpoints:

- **Devices**: Create, Get, Delete
- **Links**: Create, Get, Delete
- **Sites**: Create, Get, Delete
- ListTagsForResource

https://docs.aws.amazon.com/networkmanager/latest/APIReference/API_GetDevices.html
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/networkmanager/client/get_sites.html

### CODE REVIEW

**Comment by zkarpinski** [comment]
@bblommers I can't figure out why all my **Get** features aren't working as intended. The request only picks up the first element instead of the entire list. 

Per the documentation, the ids should be a list.
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/networkmanager/client/get_sites.html

File: moto/networkmanager/responses.py [hunk 1]
```diff
@@ -109,3 +109,150 @@ def describe_global_networks(self) -> str:
...
+        site = self.networkmanager_backend.delete_site(
+            global_network_id=global_network_id,
+            site_id=site_id,
+        )
+        return json.dumps(dict(Site=site.to_dict()))
+
+    def get_sites(self) -> str:
+        params = self._get_params()
+        global_network_id = unquote(self.path.split("/")[-2])
+        site_ids = params.get("siteIds")
```
Comment (bblommers) [inline moto/networkmanager/responses.py:139]: Looks like the `_get_params()` -method isn't quite smart enough to understand this usecase @zkarpinski.

You can directly get these values from the querystring, that will show all values:
```
self.querystring.get("siteIds")
```

Comment (zkarpinski) [reply moto/networkmanager/responses.py]: Thanks @bblommers ! I made the corrections.

File: moto/networkmanager/models.py [hunk 2]
```diff
@@ -195,6 +360,186 @@ def describe_global_networks(
...
+    ) -> List[Link]:
+        # TODO: Implement filtering by site_id, type, provider
+        gn_links = self.links.get(global_network_id) or {}
+        queried = []
+        if not link_ids:
+            queried = list(gn_links.values())
+        elif isinstance(link_ids, List) and len(link_ids) == 1:
+            link_id = link_ids[0]
+            if link_id not in gn_links:
+                raise ResourceNotFound(link_id)
```
Comment (bblommers) [inline moto/networkmanager/models.py:458]: Is this error only raised when a single `link_id` is supplied? I'd would expect the logic to be the same regardless of how many `link_ids` are supplied.

There are three possible logic paths here of course, I don't know what AWS does:
 - error only if none of the provided `link_ids` are correct
 - error if a single link is found, and show that ID
 - error when some links are not found, and supply all `link_ids` that are not found

File: tests/test_networkmanager/test_networkmanager.py [hunk 3]
```diff
@@ -191,3 +205,331 @@ def test_describe_global_networks():
...
+    site_ids = []
+    for i in range(NUM_SITES):
+        site_id = client.create_site(
+            GlobalNetworkId=gn_id,
+            Description="Test site #{i}",
+        )["Site"]["SiteId"]
+        site_ids.append(site_id)
+    sites_to_get = site_ids[0:NUM_TO_TEST]
+    resp = client.get_sites(GlobalNetworkId=gn_id, SiteIds=sites_to_get)["Sites"]
+    assert len(resp) == NUM_TO_TEST  # Fix this - Fails if greater than 1
```
Comment (bblommers) [inline tests/test_networkmanager/test_networkmanager.py:258]: Bit of a nitpick - but these comments (and the ones below) can be removed now, I think?

**Review by bblommers — CHANGES_REQUESTED** [review]
Hi @zkarpinski, thanks for the PR! This looks OK overall - just a few minor comments.

File: moto/networkmanager/models.py [hunk 4]
```diff
@@ -24,6 +24,24 @@
         "limit_default": 100,
         "unique_attribute": "core_network_arn",
     },
+    "get_sites": {
```
Comment (bblommers) [inline moto/networkmanager/models.py:27]: Good to see pagination supported from the start!

File: moto/networkmanager/models.py (hunk 2 above)
Comment (zkarpinski) [reply moto/networkmanager/models.py:458]: Tested it, it'll return an empty list. No error. Fixing logic

**Comment by zkarpinski** [comment]
@bblommers Resolved the feedback & updated exception logic.

**Review by bblommers — APPROVED** [review]
LGTM!

**Pull request merged**